### PR TITLE
Native validation update

### PR DESCRIPTION
**Pull request by RossGoncharuk**
Updated native request validation to match GO version's behavior.

### CODE REVIEW

File: src/main/java/org/prebid/server/validation/RequestValidator.java [hunk 1]
```diff
@@ -613,18 +619,23 @@ private void validateNativePlacementType(Integer placementType, int index) throw
...
         return updatedAssets;
     }
 
     private void validateNativeAsset(Asset asset, int impIndex, int assetIndex) throws ValidationException {
-        if (asset.getId() != null) {
-            throw new ValidationException("request.imp[%d].native.request.assets[%d].id must not be defined. Prebid"
-                    + " Server will set this automatically, using the index of the asset in the array as the ID",
-                    impIndex, assetIndex);
-        }
-
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/validation/RequestValidator.java]: I know that's not your code, but could you also think about changing the 
line 633
`final long assetsCount = Stream.of(title, image, video, data)` 
to
`ObjectUtils.anyNotNull(....)`

File: src/main/java/org/prebid/server/validation/RequestValidator.java [hunk 2]
```diff
@@ -613,18 +619,23 @@ private void validateNativePlacementType(Integer placementType, int index) throw
         for (int i = 0; i < assets.size(); i++) {
             final Asset asset = assets.get(i);
             validateNativeAsset(asset, impIndex, i);
-            updatedAssets.add(asset.toBuilder().id(i).build());
+
+            final Asset updatedAsset = asset.getId() != null ? asset : asset.toBuilder().id(i).build();
```
Comment (DGarbar) [inline src/main/java/org/prebid/server/validation/RequestValidator.java]: In Go, they don't set any id if there are Asset with id. 
Example (0 in go == null in Java)
GO ids: 0, 0, 0, 10 will be exception on the second id. 
 Java ids: null, null, null, 10 will be 0, 1, 2, 10

I don't really know, if it's a bug. Bc they don't have any distinction btw 0 and null. 
Probably, we can just check, if there more than one null. And if there only one empty id, we just replace it with 0, if we want to have similar behavior. But probably, it is better to throw an exception, bc it is not expected behavior.

Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/validation/RequestValidator.java]: That's what I was struggling with - what if we get the value `0`? It's either we treat `0` as both `null` and actual value, OR check for `null` only. I've decided to go with first one after talking to @rpanchyk . Let's see what he thinks on it again.

File: src/main/java/org/prebid/server/validation/RequestValidator.java (hunk 1 above)
Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/validation/RequestValidator.java]: We actually need a stream to count the amount, so if there's more than one we throw validation exception.

File: src/main/java/org/prebid/server/validation/RequestValidator.java (hunk 2 above)
Comment (rpanchyk) [reply src/main/java/org/prebid/server/validation/RequestValidator.java]: Yeah, that's the thing we always doubt in :)
So, i'll accept your decision.
PS. Java-way: check only  for `null`, but to Go-compatible we should check for both: `null` and `0`. Usually it depends on data we deal with.

**Pull request merged**